File: Types/PlacePickerOptions.swift
//
//  Options.swift
//  react-native-place-picker
//
//  Created by b0iq on 10/11/2022.
//

import Foundation

struct PlacePickerOptions: Codable {
    var title: String = "Choose Place"
    var searchPlaceholder: String = "Search..."
    var color: String = "#FF0000"
    var contrast: String = "#FFFFFF"
    var locale: String = "en-US"
    var initialCoordinates: PlacePickerCoordinate = PlacePickerCoordinate(latitude: 25.2048, longitude: 55.2708)
    var enableGeocoding = true
    var enableSearch = true
    var enableUserlocation = true
    var enableLargeTitle = true
}

File: android/src/main/java/com/placepicker/PlacePickerOptions.kt
package com.placepicker

const val PICK_OPTIONS = "com.placepicker.PICKER_OPTIONS"

enum class PlacePickerPresentation(val type: String) {
  modal("modal"),
  fullscreen("fullscreen")
}

data class PlacePickerCoordinate (
  var latitude: Double = 25.2048,
  var longitude: Double = 55.2708
)

data class PlacePickerAddress(
  var name: String? = null,
  var streetName: String? = null,
  var city: String? = null,
  var state: String? = null,
  var zipCode: String? = null,
  var country: String? = null,
)
data class PlacePickerOptions(
  val presentationStyle: PlacePickerPresentation? = PlacePickerPresentation.modal,
  val title: String = "Choose Place",
  val searchPlaceholder: String = "Search...",
  val color: String = "#FF0000",
  val contrastColor: String = "#FFFFFF",
  val locale: String = "en-US",
  val initialCoordinates: PlacePickerCoordinate = PlacePickerCoordinate(),
  val enableGeocoding: Boolean = true,
  val enableSearch: Boolean = true,
  val enableUserLocation: Boolean = true,
  val enableLargeTitle: Boolean = true,
  val rejectOnCancel: Boolean = true
)


data class PlacePickerResult (
  var coordinate: PlacePickerCoordinate? = null,
  var address: PlacePickerAddress? = null,
  var didCancel: Boolean = false,
)
